### PR TITLE
cryptonote_protocol: don't serialize an unneeded new block message

### PR DESCRIPTION
**Pull request by moneromooo-monero**
Non fluffy block nodes should now be very rare

### CODE REVIEW

**Comment by Gingeropolous** [comment]
but if someone loads the daemon with -no-fluffy-blocks (for whatever reason), will this PR affect that?

Though I guess, by now, if there were any problems with fluffyblocks we would have seen them.

**Comment by moneromooo-monero** [comment]
It should then serialize the non-fluffy version and send it. The patch doesn't remove the ability to do so, it just delays it until it knows it's actually needed.

File: src/cryptonote_protocol/cryptonote_protocol_handler.inl [hunk 1]
```diff
@@ -1667,14 +1667,7 @@ skip:
   {
     NOTIFY_NEW_FLUFFY_BLOCK::request fluffy_arg = AUTO_VAL_INIT(fluffy_arg);
     fluffy_arg.current_blockchain_height = arg.current_blockchain_height;    
-    std::vector<blobdata> fluffy_txs;
     fluffy_arg.b = arg.b;
-    fluffy_arg.b.txs = fluffy_txs;
```
Comment (stoffu) [inline src/cryptonote_protocol/cryptonote_protocol_handler.inl]: Removing this line would serialize the entire block entry into the fluffy blob, wouldn't it?

Comment (moneromooo-monero) [reply src/cryptonote_protocol/cryptonote_protocol_handler.inl]: Oops, I misread.

**Pull request merged**